src/nvim/main.c

```diff
@@ -72,6 +72,10 @@
 #include "nvim/api/private/helpers.h"
 #include "nvim/api/private/handle.h"
 
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "mbyte/latin.h.generated.h"
+#endif
```

Comment:
No, this *must not* be included. `latin.h.generated.h` does *not* contain necessary `#include`s which defines things like `char_u`, they were always supposed to be written in corresponding `.h` file. By allowing such code you add *yet another* set of headers that cannot be used on their own. Ideal situation is when all headers can be included without the errors regardless of absence/presence of other headers’ includes, but this currently is not the case for many headers.